src/Components/Blazor/Build/test/RenderingRazorIntegrationTest.cs

```diff
@@ -290,37 +290,6 @@ public void SupportsDataDashAttributes()
                 frame => AssertFrame.Attribute(frame, "data-def", "Expression value", 2));
         }
 
-        [Fact]
-        public void SupportsAttributesWithEventHandlerValues()
-        {
-            // Arrange/Act
-            var component = CompileToComponent(
-                @"<elem attr=@MyHandleEvent />
-                @code {
-                    public bool HandlerWasCalled { get; set; } = false;
-
-                    void MyHandleEvent(Microsoft.AspNetCore.Components.UIEventArgs eventArgs)
-                    {
-                        HandlerWasCalled = true;
-                    }
-                }");
```

Comment:
@SteveSandersonMS - speakup if you have concerns. I'm specifically removing support for things like this (not using directive attributes). Should we do this with `Action` and `Func<Task>` as well?

Reply:
I expect this is fine, including removing the `Action`/`Func<Task>` variants too. The *ideal*, if we have time for it, would be issuing a compile-time warning for people who put attributes named `onX` with delegate-type values when they fail to use the directive attribute.

I just checked that our extensibility for event tag helpers works nicely (it does), so people are free to define their own custom event type names and values. This is essential because the HTML spec continues to evolve, and new or nonstandard event names will keep appearing.

Given that the tag helper extensibility works well, there's no use case for people trying to define events handlers any other way. The only risk is that they don't know about how to do it, which is where a compiler warning (perhaps with a fwlink to docs) would be ideal.

Reply:
> If we don't have time to add the warning now, it could be added in the future

I'm not certain how this would work in practice. We don't know in the Razor compiler what kind of value you're passing in when you set an attribute.